astacus/coordinator/coordinator.py

```diff
@@ -239,6 +239,20 @@ async def run(self):
                 await asyncio.gather(*relock_tasks, return_exceptions=True)
             await self.request_unlock_from_nodes(locker=self.locker)
 
+    def set_status(self, state: op.Op.Status, *, from_state: Optional[op.Op.Status] = None) -> bool:
+        changed = super().set_status(state=state, from_state=from_state)
+        self._update_running_stats()
+        return changed
+
+    def _update_running_stats(self):
+        if self.stats is not None:
+            if self.info.op_status in {op.Op.Status.done, op.Op.Status.fail}:
+                op_running = 0
+            else:
+                op_running = 1
```

Comment:
I'd rather have op_running_for metric, which would have number that is ~
`int(time.monotonic()-start_time_time_monotonic+1)`.

With this, we can create alerting which is e.g. `astacus_op_running_for > 3600` to detect ops that take over hour without any hard lifting on the alerting side.

( Disclaimer: If this case is somehow covered elsewhere that's fine too, I just didn't notice it )

Reply:
True. I initially had that, but then went for the "binary" approach with just 1 or 0. I'll fix this.